fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreDeleteWhereTest.java

```diff
@@ -260,6 +562,51 @@ void testDeleteWhereSingleTypeWithFunctionIndexSplit() throws Exception {
         }
     }
 
+    @Test
+    void testDeleteWhereSingleTypeEmptyPredicate() throws Exception {
```

Comment:
Inspired by #1585 should there be a test where the entire key is grouped. eg:
```
new Index("MyRecord$sum_num",
                 new GroupingKeyExpression(field("header").nest(concatenateFields("num")), 1),
                 IndexTypes.SUM)
```
Which should fail if deleting by `num` or `path`, but should on fail if deleting by `null`